src/ImageLoaders/MzExe/PharLapExtender.cs

```diff
@@ -64,23 +76,50 @@ public override Program Load(Address addrLoad)
                     {
                         us &= 0x7FFF;
                         rdr.TryReadByte(out var b);
-                        for (int i = 0; i < us; ++i)
+                        if (b > us)
                         {
-                            w.WriteByte(b);
+                            throw new NotImplementedException();  // Corrupt file
```

Comment:
If this is the case, use `BadImageFormatException`.

Reply:
Added error handling while loading file